pkgs/by-name/wa/wallpaper-compat-hook/setup-hook.sh

```diff
@@ -0,0 +1,76 @@
+wallpaperCompatHook() {
+  echo "Running wallpaperCompatHook"
+
+  local backgrounds_dir="$out/share/backgrounds"
+  local wallpapers_dir="$out/share/wallpapers"
+
+  # Default name, if there isn't already one that we can use
+  local default_wallpaper_name=$(echo "$pname" | sed 's/wallpapers\?//i' | sed 's/^-//' | sed 's/-$//')
+  # Detect if there is an existing directory in $backgrounds_dir that we can co-opt. If there are multiple, keep the default name.
+  local existing_wallpaper_dirs=($(find "$backgrounds_dir" -mindepth 1 -maxdepth 1 -type d))
```

Comment:
```suggestion
  local existing_wallpaper_dirs=("$backgrounds_dir"/*/))
```

Reply:
For this and the below, no, because the file doesn't exist because the directory was only just created. If the directory existed, this code path wouldn't be run.

Reply:
I don't understand your point, this glob expansion and the `find` invocation should be equivalent, while handling spaces etc... better and not spawning an external command.

Reply:
Oops responded to the wrong comment. Sorry!